stdlib/public/core/CocoaArray.swift

```diff
@@ -47,23 +47,24 @@ internal struct _CocoaArrayWrapper : Collection {
   /// implementation of countByEnumeratingWith.
   func contiguousStorage(
     subRange: Range<Int>
-  ) -> UnsafeMutablePointer<AnyObject>
+  ) -> UnsafeMutablePointer<AnyObject>?
   {
     var enumerationState = _makeSwiftNSFastEnumerationState()
+    var dummyBuffer = [AnyObject]()
 
     // This function currently returns nil unless the first
     // subRange.endIndex items are stored contiguously.  This is an
     // acceptable conservative behavior, but could potentially be
     // optimized for other cases.
     let contiguousCount = withUnsafeMutablePointer(&enumerationState) {
-      self.buffer.countByEnumeratingWith($0, objects: nil, count: 0)
+      self.buffer.countByEnumeratingWith(
+        $0, objects: &dummyBuffer, count: 0)
```

Comment:
It is not a part of Fonudation, it is declared in stdlib/public/core/ShadowProtocols.swift.